common/lib/dependabot/dependency_updaters/base.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module DependencyUpdaters
+    class Base
+      def update(dependency:, requirements:, configuration:)
+        raise NotImplementedError
```

Comment:
Would it be useful to create an adapter implementation that would allow taking our existing ecosystem updaters and fitting it into this interface? If we could quickly get something going end to end with a working update it could help us discover if there's anything additional we need to pass in or return to get a successful PR.

Reply:
Yeah I think that's definitely something we should explore, it _seems_ pretty feasible to retrofit our existing implementations